src/ListTabulator/index.ts

```diff
@@ -445,274 +489,365 @@ export default class ListTabulator {
     }
 
+    const restore = saveCaret();
+
+    parentItem.after(item);
+
+    restore();
+
     /**
-     * Get content from caret till the end of the block to move it to the new item
+     * If previous parent's children list is now empty, remove it.
      */
-    const endingFragment = Caret.extractFragmentFromCaretPositionTillTheEnd();
-    if (!endingFragment) {
+    const parentItemChildWrapper = getItemChildWrapper(parentItem);
+
+    if (!parentItemChildWrapper) {
       return;
     }
-    const endingHTML = Dom.fragmentToString(endingFragment);
+
+    removeChildWrapperIfEmpty(parentItemChildWrapper);
+  }
+
+  /**
+   * Method that is used for list splitting and moving trailing items to the new separated list
+   * @param item - current item html element
+   */
+  splitList(item: ItemElement): void {
+    const currentItemChildrenList = getChildItems(item);
 
     /**
-     * Get the target item content element
+     * First child item should be unshifted because separated list should start
+     * with item with first nesting level
      */
-    if (!targetItem) {
-      return;
+    if (currentItemChildrenList !== null) {
+      const firstChildItem = currentItemChildrenList[0];
+
+      this.unshiftItem(firstChildItem);
     }
-    const targetItemContent = targetItem.querySelector<HTMLElement>(
-      `.${DefaultListCssClasses.itemContent}`
-    );
 
     /**
-     * Set a new place for caret
+     * Get trailing siblings of the current item
      */
-    if (!targetItemContent) {
+    const newListItems = getSiblings(item);
+
+    if (newListItems === null) {
       return;
     }
-    Caret.focus(targetItemContent, false);
 
     /**
-     * Save the caret position
+     * Render new wrapper for list that would be separated
      */
-    this.caret.save();
+    const newListWrapper = this.renderer!.renderWrapper(true);
 
     /**
-     * Update target item content by merging with current item html content
+     * Append new list wrapper with trailing elements
      */
-    targetItemContent.insertAdjacentHTML('beforeend', endingHTML);
+    newListItems.forEach((item) => {
+      newListWrapper.appendChild(item);
+    })
+
+    const newListContent = this.save(newListWrapper);
 
     /**
-     * Get the sublist first-level items for current item
+     * Get current list block index
      */
-    let currentItemSublistItems: NodeListOf<Element> | Element[] =
-      currentItem.querySelectorAll(
-        `.${DefaultListCssClasses.itemChildren} > .${DefaultListCssClasses.item}`
-      );
+    const currentBlock = this.block;
+
+    const currentBlockIndex = this.api.blocks.getCurrentBlockIndex()
 
     /**
-     * Create an array from current item sublist items
+     * Insert separated list with trailing items
+     * Insertion will be applied after paragraph block inserted in getOutOfList method
+     * this is why we need to increase currentBlock index by 1 (current block index is index of the paragraph block)
```

Comment:
> Insertion will be applied after paragraph block inserted in getOutOfList method this is why we need to increase currentBlock index by 1 (current block index is index of the paragraph block
 
 this part seems not actual and unclear. I'd suggest to remove it